### PR TITLE
v6

### PR DESCRIPTION
**Pull request by RoinujNosde**

### CODE REVIEW

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+        }
+    }
+
+    protected void playDeathSound(@NotNull Warrior victim) {
+        Stream<Player> players = getPlayerParticipantsStream();
+        if (!getConfig().isGroupMode()) {
+            players.forEach(p -> SoundUtils.playSound(ENEMY_DEATH, plugin.getConfig(), p));
+            return;
+        }
+        GroupManager groupManager = plugin.getGroupManager();
```
Comment (Tomut0) [inline src/main/java/me/roinujnosde/titansbattle/BaseGame.java:435]: Group**s**Manager? 👀

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+                if (!command.contains("%player%")) { // Runs the command once when %player% is not used
+                    Bukkit.dispatchCommand(Bukkit.getConsoleSender(), command);
+                    break;
+                }
+                Bukkit.dispatchCommand(Bukkit.getConsoleSender(), command.replace("%player%", player.getName()));
+            }
+        }
+    }
+
+    protected void teleport(@NotNull Collection<Warrior> collection, @NotNull Location destination) {
```
Comment (Tomut0) [inline src/main/java/me/roinujnosde/titansbattle/BaseGame.java:509]: `@NotNull Collection<Warrior> warriors`?

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+
+    protected void teleportToArena(List<Warrior> warriors) {
+        Location entrance1 = getConfig().getArenaEntrance(1);
+        Location entrance2 = getConfig().getArenaEntrance(2);
+        if (entrance2 == null) {
+            teleport(warriors, entrance1);
+            return;
+        }
+        if (getConfig().isGroupMode()) {
+            List<Group> groups = warriors.stream().map(Warrior::getGroup).distinct().collect(Collectors.toList());
```
Comment (Tomut0) [inline src/main/java/me/roinujnosde/titansbattle/BaseGame.java:525]: > map(Warrior::getGroup).distinct()

Are it's possible to be in different groups at the same time? 🤔

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 4]
```diff
@@ -0,0 +1,707 @@
...
+        private ChatColor getColor() {
+            ChatColor color;
+            if (timer > 6) {
+                color = ChatColor.GREEN;
+            } else if (timer > 3) {
+                color = ChatColor.YELLOW;
+            } else {
+                color = ChatColor.RED;
+            }
+            return color;
```
Comment (Tomut0) [inline src/main/java/me/roinujnosde/titansbattle/BaseGame.java:693]: `ChatColor color` could equals RED by default and what about static import?

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java (hunk 1 above)
Comment (RoinujNosde) [reply src/main/java/me/roinujnosde/titansbattle/BaseGame.java:435]: ClansManager?

Also can't change it now: API.

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 5]
```diff
@@ -0,0 +1,707 @@
...
+    }
+
+    public void onJoin(@NotNull Warrior warrior) {
+        if (!canJoin(warrior)) {
+            plugin.debug(String.format("Warrior %s can't join", warrior.getName()));
+            return;
+        }
+        Player player = warrior.toOnlinePlayer();
+        if (player == null) {
+            plugin.debug(String.format("onJoin() -> player %s %s == null", warrior.getName(), warrior.getUniqueId()));
```
[the remark this answers is not in the document]
Comment (RoinujNosde) [reply src/main/java/me/roinujnosde/titansbattle/BaseGame.java:108]: Too many usages now, I might change it in the future.

File: src/main/java/me/roinujnosde/titansbattle/BaseGame.java [hunk 6]
```diff
@@ -0,0 +1,707 @@
...
+            return;
+        }
+        if (!teleport(warrior, getConfig().getLobby())) {
+            plugin.debug(String.format("Player %s is dead: %s", player, player.isDead()), false);
+            player.sendMessage(plugin.getLang("teleport.error", this));
+            return;
+        }
+        SoundUtils.playSound(JOIN_GAME, plugin.getConfig(), player);
```
[the remark this answers is not in the document]
Comment (RoinujNosde) [reply src/main/java/me/roinujnosde/titansbattle/BaseGame.java:116]: Different types. FileConfiguration vs BaseGameConfiguration

**Pull request merged**